metriq_gym/qplatform/device.py

```diff
@@ -1,12 +1,336 @@
+        "description": "Ring topology (bipartite only for even number of qubits)",
+    },
+    "grid": {
+        "type": "grid",
+        "bipartite": True,
+        "description": "2D grid topology (always bipartite)",
+    },
+    "heavy_hex": {
+        "type": "heavy_hex",
+        "bipartite": True,
+        "description": "IBM heavy hexagon topology",
+    },
+    "all_to_all": {
+        "type": "complete",
+        "bipartite": lambda n: n <= 2,  # Only for very small systems
+        "description": "Complete graph (rarely bipartite)",
+    },
+}
+
+
+def create_topology_graph(topology_type: str, num_qubits: int) -> rx.PyGraph:
+    """Create a topology graph based on configuration.
+    
+    Args:
+        topology_type: Type of topology ('line', 'ring', 'grid', etc.)
+        num_qubits: Number of qubits
+        
+    Returns:
+        rx.PyGraph: The topology graph
+    """
+    graph = rx.PyGraph()
+    graph.add_nodes_from(range(num_qubits))
+    
+    if topology_type == "line":
+        # Linear chain: 0-1-2-3-...
+        for i in range(num_qubits - 1):
+            graph.add_edge(i, i + 1, None)
+    
+    elif topology_type == "ring":
+        # Ring: 0-1-2-3-...-0
+        for i in range(num_qubits - 1):
+            graph.add_edge(i, i + 1, None)
+        if num_qubits > 2:
+            graph.add_edge(num_qubits - 1, 0, None)
+    
+    elif topology_type == "grid":
+        # 2D grid topology
+        import math
+        side_length = int(math.sqrt(num_qubits))
+        for i in range(num_qubits):
+            row, col = divmod(i, side_length)
+            # Connect to right neighbor
+            if col < side_length - 1:
+                graph.add_edge(i, i + 1, None)
+            # Connect to bottom neighbor  
+            if row < side_length - 1:
+                graph.add_edge(i, i + side_length, None)
+    
+    elif topology_type == "all_to_all":
+        # Complete graph (use sparingly)
+        for i in range(num_qubits):
+            for j in range(i + 1, num_qubits):
+                graph.add_edge(i, j, None)
+    
+    else:
+        # Default to line topology for unknown types
+        for i in range(num_qubits - 1):
+            graph.add_edge(i, i + 1, None)
+    
+    return graph
+
+
+def get_available_simulators() -> Dict[str, str]:
+    """Get available local simulators.
+    
+    Returns:
+        Dict mapping device IDs to descriptions
+    """
+    available = {}
+    
+    # Check Qiskit Aer
+    try:
+        from qiskit_aer import AerSimulator
+        available.update({
+            "qiskit.aer.automatic": "Qiskit Aer automatic method selection",
+            "qiskit.aer.statevector": "Qiskit Aer statevector simulator",
+            "qiskit.aer.stabilizer": "Qiskit Aer stabilizer simulator", 
+            "qiskit.aer.density_matrix": "Qiskit Aer density matrix simulator",
+        })
+    except ImportError:
+        pass
+    
+    # Add other simulators here as they become available
```

Comment:
While it's good to see that other simulators can be supported, perhaps we can remove this snippet of commented code.